pkg/util/clusterlease.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"context"
+
+	coordinationv1 "k8s.io/api/coordination/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+)
+
+// SetLeaseOwnerFunc helps construct a newLeasePostProcessFunc which sets
+// a cluster OwnerReference to the given lease object
+func SetLeaseOwnerFunc(c client.Client, clusterName string) func(lease *coordinationv1.Lease) error {
+	return func(lease *coordinationv1.Lease) error {
+		// Setting owner reference needs cluster's UID. When lease is initially created, it is possible that
+		// the connection between karmada and cluster is not ready yet. So try to set
+		// owner reference every time when renewing the lease, until successful.
```

Comment:
I'm confused by the comments.
> it is possible that the connection between karmada and cluster is not ready yet.

How this possible?

Reply:
Yes, the comment is not distinct, this condition will never happen. Let me take a change.